### DISCUSSION TITLE
Only execute `getStaticPaths` once in dev mode possible?

### DISCUSSION THREAD

**Discussion by marceloverdijk** [vercel/next.js#20076] [Help]
I'm using Next.js to build a SSG website.

In development mode I run the site with `npm run dev` but it seems the index pages rerun the `getStaticPaths` on every page visit.

So e.g. going to the Employee list page `getStaticPaths` is executed which is expected.
Then when going to a Employee detail page and back to the Employee list page `getStaticPaths` is called again.
As the `getStaticPaths` takes a couple of seconds it gives a very sluggish developer experience...

Is there a way to avoid `getStaticPaths` to be executed every time

**Comment by jamesmosier** [comment]
In dev (via `process.env.NODE_ENV === 'development'`) you could chose a subset of paths to return and then use `fallback: true` to render the other pages on demand.

**Reply by marceloverdijk** [reply]
Yes, that could be an option but does not suit my use case. Thx for feedback.

**Comment by marceloverdijk** [comment]
I could cache the results from `getStaticPaths` and store in a temp file.
This is explained here as well: https://stackoverflow.com/questions/62561333/next-js-caching-requests-to-database-used-in-getstaticpaths-and-getstaticprop

(would be great if something like that could be done out of the box)
